Add and Implement Azure Devops as a new SCM Provider

eurubkov opened the pull request:
In AzureDevOps, the hierarchy is as follows:
Organization -> Project -> Repository -> Items.
Each organization can have multiple projects, so we need to either let the user pass in the project name/id or get readmes across all projects. That's up for the discussion.

Still need to figure out what to do when a request fails for any reason (log somewhere, notify someone, retry?).

I also changed the GetReadmes method to have the return type of Task.

atra2396 commented:
I would say collect all repos from all of the projects for now. We can add support for the extra layer later.

atra2396 commented on src/SourceControl/AzureDevOps/AzureDevOpsSourceControlAccessor.cs:
```diff
@@ -0,0 +1,148 @@
+            }
+        }
+
+        public class ItemsResponse
```
might as well put these in their own files since this integration is already in its own folder

eurubkov replied:
done

atra2396 commented on src/DomainObjects/Repository.cs:
```diff
@@ -6,11 +6,14 @@ namespace DomainObjects
 
         public Guid OrganizationId { get; set; }
-        public string Name { get; set; }
+        public string OrganizationName { get; set; }
```
Why denormalize this, instead of fetching it from the Organization itself? If it's not actually the Organization name then this should probably go in "CustomRepositoryInformation".

eurubkov replied:
My mistake. I didn't realize that repository wasn't like a git repository

eurubkov commented:
Ok I implemented getting all readmes across the whole organization for azure devops. One thing I don't necessarily like is HttpClient. Different resources give different opinions, but initializing a new HttpClient for each request will exhaust all sockets on the machine. So I moved it to be static. I also had to have two of them because I couldn't figure out how to dynamically change the media type.
